### PR TITLE
Switch to Maven-Publish + explicit artifactory plugin in 3.2.x

### PR DESCRIPTION
**Pull request by simonbasle**

### CODE REVIEW

File: gradle/setup.gradle [hunk 1]
```diff
@@ -14,107 +14,118 @@
...
+				withXml {
+					//set optional true for compileOnly dependencies, which are not in there by default
+					asNode().dependencies[0].each { node ->
+						def group = node.groupId.text()
+						def artifact = node.artifactId.text()
+						def version = node.version.text()
+						def isOptional = project.configurations.optional.allDependencies.any { dep -> dep.group == group && dep.name == artifact }
+						if (isOptional) {
+							node.appendNode('optional', true)
+							println "$group:$artifact:$version has been marked as optional: " + node.text()
```
Comment (bsideup) [inline gradle/setup.gradle]:
```suggestion
							println "$group:$artifact:$version has been marked as optional: ${node.text()}"
```

Comment (simonbasle) [reply gradle/setup.gradle]: actually, we don't even really need the full node text (was there for debugging)

**Comment by simonbasle** [comment]
- double check the produced artifacts in 3.2 against eg. 3.2.10.RELEASE
  - [ ] core: pom/jar/javadoc/sources/kdoc/docs.zip
  - [ ] test: pom/jar/javadoc/sources/kdoc
- [ ] forward merge the change with adaptations
- double check the produced artifacts in 3.3 against 3.3.0.RELEASE
  - [ ] core: pom/jar/javadoc/sources/docs.zip
  - [ ] no benchmarks
  - [[](url) ] test: pom/jar/javadoc/sources
  - [ ] tools: pom/jar/original/javadoc/sources

**Comment by simonbasle** [comment]
⚠️ in customVersion mode the pdf is not included in docsZip distribution

**Comment by simonbasle** [comment]
I now have fixed the above issue but found another one:
⚠️ the packaging of the `pom.xml` is wrongly explicitely set to `pom` 🤒

**Comment by simonbasle** [comment]
I have fixed both issues above, will now merge and arrange the forward merge

**Pull request merged**